base/buildflag_config.h

```diff
@@ -0,0 +1,44 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BASE_BUILDFLAG_CONFIG_H_
+#define BRAVE_BASE_BUILDFLAG_CONFIG_H_
+
+#include <string>
+#include <string_view>
+
+#include "build/buildflag.h"
+
+#define BUILDFLAG_CONFIG(flag) \
```

Comment:
With the name `BUILDFLAG_CONFIG`, it isn't super-obvious what it's for or when to use it.

Since this only works for strings, and it always returns strings, would something like `BUILDFLAG_STRING` be an improvement?

Reply:
I was trying to differentiate between buildflags that are used for guards, etc... and buildflags that we use for config. There are other string buildflags that would not make sense to override so not sure if `BUILDFLAG_STRING` is right, but I'm open to other suggestions. I guess it does sound a little like some kind of config for buildflags vs a buildflag that is a config var, but I'm struggling to come up with something better :)

Reply:
I definitely need to add some comments here either way